include/swift/AST/Decl.h

```diff
@@ -5958,6 +5958,9 @@ class FuncDecl : public AbstractFunctionDecl {
   bool isConsuming() const {
     return getSelfAccessKind() == SelfAccessKind::__Consuming;
   }
+  bool isCallFunction() const {
+    return getName().str() == "callFunction" && isInstanceMember();
```

Comment:
Todo: unify "call function"/"call method" terminology, and use one spelling consistently.

Reply:
Any plans to diagnose static or class callFunction?

Reply:
What diagnostic do you have in mind?

---

Reply:
Precisely. Thanks.

Reply:
> A detail is that the current diagnostics are being ported from the old system (CSDiag.cpp) to the new request-based system, so adding diagnostics to the old system might be throwaway work. 

You can actually add diagnostics to a new diagnostic framework right now, it's live, in fact me and couple of others have ported a lot of diagnostics already, see https://github.com/apple/swift/pull/26074 for the most recent example of how easy it is to add new diagnostics now.